mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -259,18 +255,6 @@ public ConfigBuilder skipResidualFiltering() {
     return splits;
   }
 
-  private static void checkResiduals(CombinedScanTask task) {
-    task.files().forEach(fileScanTask -> {
```

Comment:
Seems like we should still do this check for Pig and Hive as we don't have residual evaluators for these. If `IcebergInputFormat.ConfigBuilder#skipResidualFiltering` is `true` then this check should be there for Pig and Hive rows

Reply:
If the `skipResidualFiltering` is true, we skip the evaluation and thus don't need `checkResiduals`. How about we set `skipResidualFiltering` to true as default?

Reply:
Residual filtering is specific to the object model. That's why the cast to `Record` works.

I agree with @rdsr that if residuals should be applied (not skipped) then we need to either ensure that generics are used, or that there are no residuals. So the check should still happen when `applyResiduals` is true and the in-memory model is Pig or Hive.